### PR TITLE
Next v2.4.0

### PR DESCRIPTION
**Pull request by diegoalzate**
## New Features
- multiple group dimensions
- hearts multiplier
- fix edge cases of plural score calculation
- test coverage of votes service
- link support
- configuration improvements (e.g. github actions)
- quadratic score calculation added to result page statistics

### CODE REVIEW

File: src/db/events.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ export const events = pgTable('events', {
   id: uuid('id').primaryKey().defaultRandom(),
   name: varchar('name').notNull(),
   description: varchar('description'),
+  eventLink: varchar('event_link'),
```
Comment (diegoalzate) [inline src/db/events.ts:10]: we should rename this just to link, i think we should stop prepending attributes with table name. Not really consistent throughout the tables which can be confusing. For example we do not say description is eventDescription. This is something that slipped through the cracks and we have an issue with this throughout the schema

File: src/services/users.ts [hunk 2]
```diff
@@ -54,9 +65,86 @@ export function getUserAttributes(dbPool: PostgresJsDatabase<typeof db>) {
...
+
+  return null;
+}
+
+/**
+ * Upserts user data in the database.
+ * @param {PostgresJsDatabase<typeof db>} dbPool - The database connection pool.
+ * @param {string} userId - The ID of the user to update.
+ * @param {UserData} userData - The updated user data.
+ * @returns {Promise<any>} - A promise resolving to the updated user data.
```
Comment (diegoalzate) [inline src/services/users.ts:114]: lets remove this <any> in the tsdoc

File: src/services/users.ts [hunk 3]
```diff
@@ -54,9 +65,86 @@ export function getUserAttributes(dbPool: PostgresJsDatabase<typeof db>) {
...
 }
 
+/**
+ * Checks user data for existing entries in the database.
+ * @param {PostgresJsDatabase<typeof db>} dbPool - The database connection pool.
+ * @param {string} userId - The ID of the user to check.
+ * @param {UserData} userData - The user data to check.
+ * @returns {Promise<Array<string> | null>} - An array of errors if user data conflicts, otherwise null.
+ */
+async function checkUserData(
```
Comment (diegoalzate) [inline src/services/users.ts:75]: i think a rename here to `validateUserData` can give us a better overview on what is going on inside the function

**Review by diegoalzate — COMMENTED** [review]
left some comments for the first review

File: src/db/events.ts (hunk 1 above)
Comment (diegoalzate) [reply src/db/events.ts:10]: i think the start of this was forumQuestions sadly

File: src/services/users.ts (hunk 2 above)
Comment (diegoalzate) [reply src/services/users.ts:114]: either we leave let typescript infer the return value or we write what it actually returns but the any is worse for a user of this function

**Review by MartinBenediktBusch — APPROVED** [review]
